lang/syn/src/codegen/accounts/to_account_metas.rs

```diff
@@ -22,26 +22,30 @@ pub fn generate(accs: &AccountsStruct) -> proc_macro2::TokenStream {
             if is_optional {
                 quote! {
                     if let Some(#name) = &self.#name {
-                        account_metas.extend(#name.to_account_metas(#is_signer));
+                        account_metas.push(#name.to_account_meta(#is_signer));
                     } else {
                         account_metas.push(AccountMeta::new_readonly(crate::ID, false));
                     }
                 }
             } else {
                 quote! {
-                    account_metas.extend(self.#name.to_account_metas(#is_signer));
+                    account_metas.push(self.#name.to_account_meta(#is_signer));
```

Comment:
yeh, this basically breaks how composite accounts work. Since you're calling `to_account_metas` on the another account context type which has itself used the `derive(Accounts)` macro.

Reply:
There may be a way to handle this by checking if the type is a composite account.

Alternatively, would be nice if the `derive(Accounts)` macro also implemented `Iterator` and iterated through the accounts in order. This could allow you to call `to_account_meta` on each element of a composite account.